File: pinyougou-interface/src/main/java/cn/itcast/core/service/BrandService.java
package cn.itcast.core.service;

import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.good.Brand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    public List<Brand> query();

    PageResult queryPage(int pageNum, int pageSize);

    void add(Brand brand) throws Exception;

    Brand queryBrandById(long id)throws Exception;

    void update(Brand brand);

    void deleteByIds(Long[] ids);

    PageResult queryPageWithSearch(int pageNum, int pageSize, Brand brand);

    List<Map> selectOptionList();

    void updateStatus(Long[] ids, String status) throws Exception;

    void updateStatus1(Long[] ids, String status)throws Exception;
}

File: pinyougou-web-shop/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    //查询所有
    this.findAll=function () {
        return $http.get("../brand/query.do");
    }

    //不带条件的查询
    this.queryPage=function (currentPage,itemsPerPage) {
        return $http.get("../brand/queryPage.do?pageNum=" + currentPage +
            "&pageSize=" + itemsPerPage)
    }

    //带条件的查询
    this.search=function (currentPage,itemsPerPage,searchEntity) {
        return $http.post("../brand/queryPageWithSearch.do?pageNum=" + currentPage +
            "&pageSize=" + itemsPerPage, searchEntity);
    }

    //添加
    this.add=function (entity) {
        return $http.post("../brand/add.do", entity)
    }

    //更新
    this.update=function (entity) {
        return $http.post("../brand/update.do", entity)
    }


    //根据Id查询
    this.queryBrandById=function (id) {
        return $http.get("../brand/queryBrandById.do?id=" + id);
    }

    //删除
    this.deleteByIds=function (checkedIds) {
       return $http.get("../brand/deleteByIds.do?ids=" + checkedIds)
    }

    //查询所有返回List<map>集合
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do")
    }

    this.updateStatus = function(ids,status){
        return $http.get('../brand/updateStatus.do?ids='+ids+"&status="+status);
    }

})